chatsky/llm/example_selector.py

```diff
@@ -0,0 +1,117 @@
+"""
+Example selection.
+----------
+This module provides support for example guided generation.
+"""
+
+from typing import List, Dict
+
+from pydantic import BaseModel, Field
+
+from langchain_core.example_selectors.base import BaseExampleSelector
+from langchain_core.messages import HumanMessage, AIMessage
+
+
+class Example(BaseModel):
+    '''How to answer a user.'''
+
+    example_input: str = Field(description="An example of the user's query.")
+
+    example_output: str = Field(description="A desired answer example")
+
+
+class ExampleSelector(BaseModel, arbitrary_types_allowed=True):
+    '''Class for example selection'''
+
+    examples: List[Dict[str, str] | Example] 
```

Comment:
If we make this `List[Example]` then dictionaries will be cast to `Example` instances when initializing this `ExampleSelector.